src/node_platform.cc

```diff
@@ -0,0 +1,183 @@
+#include "node_platform.h"
+
+#include "util.h"
+
+namespace node {
+
+using v8::Isolate;
+using v8::Platform;
+using v8::Task;
+using v8::TracingController;
+
+static void FlushTasks(uv_async_t* handle) {
+  NodePlatform* platform = static_cast<NodePlatform*>(handle->data);
+  platform->FlushForegroundTasksInternal();
+}
+
+static void BackgroundRunner(void* data) {
+  TaskQueue<Task>* background_tasks = static_cast<TaskQueue<Task>*>(data);
+  while (Task* task = background_tasks->BlockingPop()) {
+    task->Run();
+    delete task;
+    background_tasks->NotifyOfCompletion();
+  }
+}
+
+NodePlatform::NodePlatform(int thread_pool_size, uv_loop_t* loop,
+                           TracingController* tracing_controller)
+    : loop_(loop) {
+  CHECK_EQ(0, uv_async_init(loop, &flush_tasks_, FlushTasks));
+  flush_tasks_.data = static_cast<void*>(this);
+  uv_unref(reinterpret_cast<uv_handle_t*>(&flush_tasks_));
+  if (tracing_controller) {
+    tracing_controller_.reset(tracing_controller);
+  } else {
+    TracingController* controller = new TracingController();
+    tracing_controller_.reset(controller);
+  }
+  for (int i = 0; i < thread_pool_size; i++) {
+    uv_thread_t* t = new uv_thread_t();
+    if (uv_thread_create(t, BackgroundRunner, &background_tasks_) != 0) {
+      delete t;
+      break;
+    }
+    threads_.push_back(std::unique_ptr<uv_thread_t>(t));
+  }
+}
+
+void NodePlatform::Shutdown() {
+  background_tasks_.Stop();
+  for (size_t i = 0; i < threads_.size(); i++) {
+    uv_thread_join(threads_[i].get());
+  }
+  uv_close(reinterpret_cast<uv_handle_t*>(&flush_tasks_), nullptr);
```

Comment:
I'm afraid not.  I suppose that just leaves what you currently have.  I'd remove the call to `uv_close()` since it's pointless and replace it with a big fat comment.  A similar comment should be added in src/node.cc.